endpoint.go

```diff
@@ -340,23 +408,26 @@ func (ep *endpoint) sbLeave(sbox Sandbox, options ...EndpointOption) error {
 
 	ep.processOptions(options...)
 
+	n, err := ep.getNetworkFromStore()
```

Comment:
To be consistent with other APIs, can you please change Leave() to make sure we get the endpoint from the store before calling sbLeave() ?
That way it will be consistent with other APIs that we dont trust the calling receiver other than to make use of the ID.